test/e2e/tests/request-queuing/switchChain-watchAsset.spec.js

```diff
@@ -82,16 +75,21 @@ describe('Request Queue SwitchChain -> WatchAsset', function () {
           tag: 'button',
         });
 
-        await switchToNotificationWindow(driver);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
 
         // Confirm Switch Network
-        await driver.findClickableElement({
-          text: 'Confirm',
-          tag: 'button',
+        const switchEthereumChainRequestTwo = JSON.stringify({
+          jsonrpc: '2.0',
+          method: 'wallet_switchEthereumChain',
+          params: [{ chainId: '0x1' }],
+        });
+        await driver.executeScript(
+          `window.ethereum.request(${switchEthereumChainRequestTwo})`,
+        );
+        await driver.waitForSelector({
+          css: '[id="chainId"]',
+          text: '0x1',
         });
-        await driver.clickElement({ text: 'Confirm', tag: 'button' });
-
-        await driver.waitUntilXWindowHandles(2);
```

Comment:
We will need to retain this check as well to show that the test succeeded.

The goal of the test was to ensure that the pending "Add token" request was discarded. We checked this by ensuring the popup was not open by checking the number of windows. Perhaps there is a better way, but we still need to check this somehow.

Reply:
Aah! Thanks Mark. I have updated the test here https://github.com/MetaMask/metamask-extension/pull/27980/commits/a18a31d1c6975dccfc83ece4064ce16464663af3